src/foam/java/refinements.js

```diff
@@ -2288,9 +2288,8 @@ foam.CLASS({
         // this.fs_.writeFileSync("/tmp/" + of + ".new", this.fs_.readFileSync(outfile).toString());
 
         this.fs_.writeFileSync(outfile, javaSource);
-
-        X.javaFiles.push(outfile);
       }
+      X.javaFiles.push(outfile);
```

Comment:
Why did you move this line here? The logic is so that javaFiles only contains a list of files which have been updated, not all files.

Reply:
I wanted to recompile because I was changing the generated java code but haven't found a way to only pick the files that were changed. I'll revert this as it makes the build slower.